gtsfm/common/gtsfm_data.py

```diff
@@ -27,15 +29,32 @@ class GtsfmData:
     The situation of non-contiguous cameras can exists because of failures in front-end.
     """
 
-    def __init__(self, number_images: int) -> None:
+    def __init__(
+        self,
+        number_images: int,
+        cameras: Optional[Dict[int, PinholeCameraCal3Bundler]] = None,
+        tracks: Optional[List[SfmTrack]] = None,
+        scene_mesh: Optional[Trimesh] = None,
+    ) -> None:
         """Initializes the class.
 
         Args:
-            number_images: number of images/cameras in the scene.
+            number_images: number of putative images/cameras in the scene; cameras may be pruned during the verification
+                process.
+            cameras: cameras used to reconstruct the scene.
+            tracks: tracks in reconstucted map of the scene.
+            scene_mesh: triangular surface mesh of reconstructed scene.
         """
-        self._cameras: Dict[int, PinholeCameraCal3Bundler] = {}
-        self._tracks: List[SfmTrack] = []
         self._number_images = number_images
+        self._cameras = cameras if cameras else {}
+        self._tracks = tracks if tracks else []
+        self.scene_mesh = scene_mesh
+
+    def get_two_view_data(self, i1: int, i2: int) -> "GtsfmData":
+        """Collects GtsfmData for a single image pair."""
+        # TODO (travisdriver): also collect tracks if available.
+        cameras_pair = {0: self.get_camera(i1), 1: self.get_camera(i2)}
```

Comment:
I think we should always use `i1` and `i2` directly, instead of remapping to 0 and 1, since this can create some confusion/bugs downstream for the user.

@ayushbaid @akshay-krishnan  what are your thoughts?